streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -172,7 +172,7 @@
                                         REPLICATION_FACTOR_DOC)
                                 .define(TIMESTAMP_EXTRACTOR_CLASS_CONFIG,
                                         Type.CLASS,
-                                        ConsumerRecordTimestampExtractor.class.getName(),
+                                        FailingConsumerRecordTimestampExtractor.class.getName(),
```

Comment:
We might even keep the name as-is. The reason is, that right now, if `ConsumerRecordTimestampExtractor` returns -1, a failure occurs too. Just not in the extractor but in `RecordQueue`.